c++/src/capnp/compiler/generics.c++

```diff
@@ -131,10 +131,19 @@ bool BrandedDecl::compileAsType(
         }
 
         if (elementType.isAnyPointer()) {
-          addError(errorReporter, "'List(AnyPointer)' is not supported.");
-          // Seeing List(AnyPointer) later can mess things up, so change the type to Void.
-          elementType.setVoid();
-          return false;
+          auto unconstrained = elementType.getAnyPointer().getUnconstrained();
+
+          if (unconstrained.isAnyKind()) {
+            addError(errorReporter, "'List(AnyKind)' is not supported.");
+            // Seeing List(AnyKind) later can mess things up, so change the type to Void.
+            elementType.setVoid();
+            return false;
+          } else if (unconstrained.isStruct()) {
+            addError(errorReporter, "'List(Struct)' is not supported.");
```

Comment:
Similarly here, change `List(Struct)` to `List(AnyStruct)`, since that's what you'd actually have to write in code to trigger this error.